server/repository/src/db_diesel/temperature_log.rs

```diff
@@ -1,28 +1,50 @@
 use super::{
+    location_row::{location, location::dsl as location_dsl},
+    sensor_row::{sensor, sensor::dsl as sensor_dsl},
+    temperature_breach_row::{
+        temperature_breach, temperature_breach::dsl as temperature_breach_dsl,
+    },
     temperature_log_row::{temperature_log, temperature_log::dsl as temperature_log_dsl},
-    DBType, StorageConnection, TemperatureLogRow,
+    DBType, StorageConnection, TemperatureBreachRow, TemperatureLogRow,
+};
+use diesel::{
+    helper_types::{InnerJoin, IntoBoxed, LeftJoin},
+    prelude::*,
 };
-use diesel::prelude::*;
 
 use crate::{
     diesel_macros::{apply_date_time_filter, apply_equal_filter, apply_sort, apply_sort_no_case},
+    location::{LocationFilter, LocationRepository},
     repository_error::RepositoryError,
+    LocationRow, SensorFilter, SensorRepository, SensorRow, TemperatureBreachFilter,
+    TemperatureBreachRepository,
 };
 
 use crate::{DatetimeFilter, EqualFilter, Pagination, Sort};
 
 #[derive(PartialEq, Debug, Clone)]
 pub struct TemperatureLog {
     pub temperature_log_row: TemperatureLogRow,
+    pub sensor_row: SensorRow,
+    pub location_row: Option<LocationRow>,
+    pub temperature_breach_row: Option<TemperatureBreachRow>,
 }
 
+pub type TemperatureLogJoin = (
+    TemperatureLogRow,
+    SensorRow,
+    Option<LocationRow>,
+    Option<TemperatureBreachRow>,
+);
+
 #[derive(Clone, PartialEq, Debug)]
 pub struct TemperatureLogFilter {
     pub id: Option<EqualFilter<String>>,
-    pub sensor_id: Option<EqualFilter<String>>,
     pub store_id: Option<EqualFilter<String>>,
-    pub location_id: Option<EqualFilter<String>>,
     pub timestamp: Option<DatetimeFilter>,
+    pub sensor: Option<SensorFilter>,
+    pub location: Option<LocationFilter>,
+    pub temperature_breach: Option<TemperatureBreachFilter>,
```

Comment:
Same here, these joined rows don't seem to be used, do we really need to join them for the query?

Reply:
Same comment as above